scapy/contrib/automotive/ecu.py

```diff
@@ -418,69 +574,84 @@ class EcuAnsweringMachine(AnsweringMachine):
                                  Listen-only, responds with the main_socket.
                                  `None` to disable broadcast capabilities.
         :param basecls: Provide a basecls of the used protocol
-
-           Usage:
-           >>> resp = EcuResponse(session=range(0,255), security_level=0, responses=UDS() / UDS_NR(negativeResponseCode=0x7f, requestServiceId=0x10))  # noqa: E501
-           >>> sock = ISOTPSocket(can_iface, sid=0x700, did=0x600, basecls=UDS)  # noqa: E501
-           >>> answering_machine = EcuAnsweringMachine(supported_responses=[resp], main_socket=sock, basecls=UDS)  # noqa: E501
-           >>> sim = threading.Thread(target=answering_machine, kwargs={'count': 4, 'timeout':5})  # noqa: E501
-           >>> sim.start()
-       """
-    function_name = "EcuAnsweringMachine"
-    sniff_options_list = ["store", "opened_socket", "count", "filter", "prn", "stop_filter", "timeout"]  # noqa: E501
-
-    def parse_options(self, supported_responses=None,
-                      main_socket=None, broadcast_socket=None, basecls=Raw,
-                      timeout=None):
-        self.main_socket = main_socket
-        self.sockets = [self.main_socket]
+        :param timeout: Specifies the timeout for sniffing in seconds.
+        """
+        self.__ecu_state = EcuState(session=1)
+        self.__main_socket = main_socket  # type: Optional[SuperSocket]
+        self.__sockets = [self.__main_socket]
 
         if broadcast_socket is not None:
-            self.sockets.append(broadcast_socket)
+            self.__sockets.append(broadcast_socket)
+
+        if initial_ecu_state:
+            self.__ecu_state = initial_ecu_state
 
-        self.ecu_state = Ecu(logging=False, verbose=False,
-                             store_supported_responses=False)
-        self.basecls = basecls
-        self.supported_responses = supported_responses
+        self.__basecls = basecls  # type: Type[Packet]
+        self.__supported_responses = supported_responses
 
         self.sniff_options["timeout"] = timeout
-        self.sniff_options["opened_socket"] = self.sockets
+        self.sniff_options["opened_socket"] = self.__sockets
+
+    @property
+    def state(self):
+        # type: () -> EcuState
+        return self.__ecu_state
 
     def is_request(self, req):
-        return req.__class__ == self.basecls
+        # type: (Packet) -> bool
+        return req.__class__ == self.__basecls
```

Comment:
Fine if you're sure `isinstance(req, self.__basecls)` doesn't fit your purpose.

Reply:
Thanks for pointing this out.